### PR TITLE
Change publishToMavenLocal to publishPluginToMavenLocal only

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
Change publishToMavenLocal to publishPluginToMavenLocal only

### Issues Resolved
After this PR: https://github.com/opensearch-project/opensearch-build/pull/4434 merged, build for geospatial and opensearch-reports failed, this is to fix this issue.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Thanks @zane-neo `publishPluginZipPublicationToMavenLocal` did not break for `opensearch-reports` and `geospatial`, from what we saw in the past PR https://github.com/opensearch-project/opensearch-build/pull/4434.

**Review by peterzhuamazon — APPROVED** [review]
Approve this and will wait for @prudhvigodithi second approval and merge after his local test.

Thanks.

**Comment by prudhvigodithi** [comment]
Thanks, I have tested 2.12 build on local with this change and looks good.

**Pull request merged**
